pkg/asset/machines/gcp/gcpmachines.go

```diff
@@ -0,0 +1,182 @@
+	}
+
+	// TODO tags aren't currently being set in GCPMachine which only has
+	// AdditionalNetworkTags []string
+
+	masterSubnet := installConfig.Config.Platform.GCP.ControlPlaneSubnet
+	if masterSubnet == "" {
+		masterSubnet = gcptypes.DefaultSubnetName(infraID, "master")
+	}
+
+	gcpMachine := &capg.GCPMachine{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+			Kind:       "GCPMachine",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"cluster.x-k8s.io/control-plane": "",
+			},
+		},
+		Spec: capg.GCPMachineSpec{
+			InstanceType:     mpool.InstanceType,
+			Subnet:           ptr.To(masterSubnet),
+			AdditionalLabels: getLabelsFromInstallConfig(installConfig, infraID),
+			Image:            ptr.To(osImage),
+			RootDeviceType:   ptr.To(capg.DiskType(mpool.OSDisk.DiskType)),
+			RootDeviceSize:   mpool.OSDisk.DiskSizeGB,
+		},
+	}
+	// Set optional values from machinepool
+	if mpool.OnHostMaintenance != "" {
+		gcpMachine.Spec.OnHostMaintenance = ptr.To(capg.HostMaintenancePolicy(mpool.OnHostMaintenance))
+	}
+	if mpool.ConfidentialCompute != "" {
+		gcpMachine.Spec.ConfidentialCompute = ptr.To(capg.ConfidentialComputePolicy(mpool.ConfidentialCompute))
+	}
+	if mpool.SecureBoot != "" {
+		shieldedInstanceConfig := capg.GCPShieldedInstanceConfig{}
+		shieldedInstanceConfig.SecureBoot = capg.SecureBootPolicyEnabled
+		gcpMachine.Spec.ShieldedInstanceConfig = ptr.To(shieldedInstanceConfig)
+	}
+	if mpool.ServiceAccount != "" {
+		serviceAccount := &capg.ServiceAccount{
+			Email: mpool.ServiceAccount,
+			// Set scopes to value defined at
+			// https://cloud.google.com/compute/docs/access/service-accounts#scopes_best_practice
+			Scopes: []string{"https://www.googleapis.com/auth/cloud-platform"},
+		}
+		gcpMachine.Spec.ServiceAccount = serviceAccount
+	}
+
+	return gcpMachine
+}
+
+// Create a CAPI machine based on the CAPG machine.
+func createCAPIMachine(name string, dataSecret string, infraID string) *capi.Machine {
+	machine := &capi.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"cluster.x-k8s.io/control-plane": "",
+			},
+		},
+		Spec: capi.MachineSpec{
+			ClusterName: infraID,
+			// Leave empty until ignition support is added
+			// Bootstrap: capi.Bootstrap{
+			//	DataSecretName: ptr.To(dataSecret),
+			// },
+			InfrastructureRef: v1.ObjectReference{
+				APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+				Kind:       "GCPMachine",
+				Name:       name,
+			},
+		},
+	}
+
+	return machine
+}
+
+func getLabelsFromInstallConfig(installConfig *installconfig.InstallConfig, infraID string) map[string]string {
+	ic := installConfig.Config
+
+	userLabels := map[string]string{}
+	for _, label := range ic.Platform.GCP.UserLabels {
+		userLabels[label.Key] = label.Value
+	}
+	// add OCP default label
+	userLabels[fmt.Sprintf("kubernetes-io-cluster-%s", infraID)] = "owned"
```

Comment:
Should we consider adding the same owned tag found here: https://github.com/openshift/installer/pull/7907 ?

This can also be a common function as many cluster api resources will require these labels.

Reply:
Looks like this is being added by capg https://github.com/openshift/installer/blob/master/vendor/sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1/labels.go#L92-L119